dartagnan/src/main/java/com/dat3m/dartagnan/asserts/AssertNot.java

```diff
@@ -25,10 +26,8 @@ AbstractAssert getChild(){
 
     @Override
     public BooleanFormula encode(SolverContext ctx) {
-        if(child != null){
-            return ctx.getFormulaManager().getBooleanFormulaManager().not(child.encode(ctx));
-        }
-        throw new RuntimeException("Empty assertion clause in " + this.getClass().getName());
+    	Preconditions.checkNotNull(child, "Empty assertion clause in " + this.getClass().getName());
```

Comment:
Don't use `checkNotNull` here. It should be done in the constructor of `AssertNot`.